psamm/datasource/native.py

```diff
@@ -174,28 +174,38 @@ class NativeModel(object):
     directory subtree that specifies part of the model.
     """
 
-    def __init__(self, path):
+    def __init__(self, model_from):
         """Create a model from the specified model file or directory"""
-
-        if os.path.isfile(path):
-            self._context = FilePathContext(path)
-            with open(self._context.filepath, 'r') as f:
-                self._model = yaml_load(f)
-        else:
-            # Try to open the default file
-            for filename in DEFAULT_MODEL:
-                try:
-                    self._context = FilePathContext(
-                        os.path.join(path, filename))
-                    with open(self._context.filepath, 'r') as f:
-                        self._model = yaml_load(f)
-                        break
-                except:
-                    logger.debug('Failed to load model file', exc_info=True)
+        if type(model_from) is str:
+            if '\n' in model_from:
```

Comment:
This is unfortunately not a reliable way of checking if a string is a path or a YAML document. It is possible to create a valid YAML document that has no newline characters.